runtime/src/main/java/org/corfudb/runtime/LogReplicationLogicalGroupClient.java

```diff
@@ -0,0 +1,403 @@
+        } catch (NoSuchElementException nse) {
+            tempSourceMetadataTable = corfuStore.openTable(CORFU_SYSTEM_NAMESPACE, LR_MODEL_METADATA_TABLE_NAME, ClientDestinationInfoKey.class,
+                    DestinationInfoVal.class, null, TableOptions.fromProtoSchema(DestinationInfoVal.class));
+        }
+        this.sourceMetadataTable = tempSourceMetadataTable;
+
+        register();
+    }
+
+    /**
+     * Adds client and model to LogReplicationRegistrationTable.
+     */
+    private void register() {
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    CorfuStoreEntry<ClientRegistrationId, ClientRegistrationInfo, Message> existingRecord = txn.getRecord(replicationRegistrationTable, clientKey);
+                    ClientRegistrationInfo clientRegistrationInfo = existingRecord.getPayload();
+
+                    if (clientRegistrationInfo != null) {
+                        log.warn("Client already registered.\n--- ClientRegistrationId ---\n{}--- ClientRegistrationInfo ---\n{}", clientKey, clientRegistrationInfo);
+                    } else {
+                        txn.putRecord(replicationRegistrationTable, clientKey, clientInfo, null);
+                        txn.commit();
+                    }
+                } catch (TransactionAbortedException tae) {
+                    log.error("[{}] {}: {}", clientName, "Unable to register client", tae);
+                    throw new RetryNeededException();
+                }
+                return null;
+            }).run();
+        } catch (InterruptedException e) {
+            log.error("[{}] {}: {}", clientName, "Client registration failed", e);
+            throw new UnrecoverableCorfuInterruptedError(e);
+        }
+    }
+
+    /**
+     * Adds a destination to LogReplicationModelMetadataTable for monitoring.
+     *
+     * @param logicalGroup The group of tables to be replicated, represented as a string.
+     * @param destination The clusters which the specified group will be replicated to.
+     */
+    public void addDestination(String logicalGroup, String destination) {
+        Preconditions.checkArgument(isValid(logicalGroup), String.format("[%s] %s", clientName, "logicalGroup is null or empty."));
+        Preconditions.checkArgument(isValid(destination), String.format("[%s] %s", clientName, "destination is null or empty."));
+        try {
+            IRetry.build(ExponentialBackoffRetry.class, () -> {
+                try (TxnContext txn = corfuStore.txn(CORFU_SYSTEM_NAMESPACE)) {
+                    ClientDestinationInfoKey clientInfoKey = buildClientDestinationInfoKey(logicalGroup);
+                    CorfuStoreEntry<ClientDestinationInfoKey, DestinationInfoVal, Message> existingRecord = txn.getRecord(sourceMetadataTable, clientInfoKey);
+                    DestinationInfoVal clientDestinations = existingRecord.getPayload();
```

Comment:
nit: line 151 and 152 are also common across all methods. If you are pushing changes again, you can think of pulling them out to another method as well.